src/transaction/log_page_buffer.c

```diff
@@ -7303,6 +7304,97 @@ logpb_checkpoint (THREAD_ENTRY * thread_p)
 #undef detailed_er_log
 }
 
+/*
+ * log_checkpoint_trantable - execute a transaction table checkpoint;
+ *          wait for log pages to be persisted up to the lsa where where the transaction table
+ *          snapshot was taken; delete previous transaction table checkpoints
+ *
+ * return: error status
+ *
+ */
+int
+logpb_checkpoint_trantable (THREAD_ENTRY * const thread_p)
+{
+  const bool detailed_logging = prm_get_bool_value (PRM_ID_LOG_CHKPT_DETAILED);
+
+  if (!is_tran_server_with_remote_storage ())
+    {
+      er_log_debug (ARG_FILE_LINE, "checkpoint_trantable: only allowed on transaction server with remote storage\n");
+      return ER_FAILED;
+    }
+
+  {
+    LOG_CS_ENTER (thread_p);
+    // *INDENT-OFF*
+    scope_exit <std::function<void (void)>> unlock_log_cs_on_exit ([thread_p] ()
+    {
+      LOG_CS_EXIT(thread_p);
+    });
+    // *INDENT-ON*
+
+    cublog::checkpoint_info trantable_checkpoint_info;
+
+    if (detailed_logging)
+      {
+	_er_log_debug (ARG_FILE_LINE, "checkpoint_trantable: started, loading trantable\n");
+      }
+    LOG_LSA dummy_smallest_tran_lsa = NULL_LSA;
+    trantable_checkpoint_info.load_trantable_snapshot (thread_p, dummy_smallest_tran_lsa);
+
+    // loading the transaction table snapshot ensures also that a snapshot lsa has been set
+    const log_lsa trantable_checkpoint_lsa = trantable_checkpoint_info.get_snapshot_lsa ();
+
+    if (trantable_checkpoint_lsa.is_null ())
+      {
+	// no transaction table checkpoint can be recorded
+	er_log_debug (ARG_FILE_LINE, "checkpoint_trantable: failed; null checkpoint lsa\n");
+	return ER_FAILED;
+      }
```

Comment:
`trantable_checkpoint_lsa` cannot be null.
```suggestion
    const log_lsa trantable_checkpoint_lsa = trantable_checkpoint_info.get_snapshot_lsa ();
    assert (!trantable_checkpoint_lsa.is_null ());
```

Reply:
already asserted in `get_snapshot_lsa`